test/Microsoft.AspNet.FileProviders.Combined.Tests/CombinedFileProviderTests.cs

```diff
@@ -0,0 +1,252 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using Microsoft.AspNet.FileProviders.Combined.Tests.TestUtility;
+using Xunit;
+using System.Collections.Generic;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.AspNet.FileProviders.Embedded.Tests
+{
+    public class CombinedFileProviderTests
+    {
+        [Fact]
+        public void GetFileInfo_ReturnsNotFoundFileInfo_IfNoFileProviderSpecified()
+        {
+            // Arrange
+            var provider = new CombinedFileProvider();
+
+            // Act
+            var fileInfo = provider.GetFileInfo("DoesNotExist.txt");
+
+            // Assert
+            Assert.NotNull(fileInfo);
+            Assert.False(fileInfo.Exists);
+        }
+
+        [Fact]
+        public void GetFileInfo_ReturnsNotFoundFileInfo_IfFileDoesNotExist()
+        {
+            // Arrange
+            var provider = new CombinedFileProvider(new MockFileProvider(new MockFileInfo("DoesExist.txt")));
+
+            // Act
+            var fileInfo = provider.GetFileInfo("DoesNotExist.txt");
+
+            // Assert
+            Assert.NotNull(fileInfo);
+            Assert.False(fileInfo.Exists);
+        }
+
+        [Fact]
+        public void GetFileInfo_ReturnsTheFirstFoundFileInfo()
+        {
+            // Arrange
+            var fileName = "File1";
+            var expectedFileInfo = new MockFileInfo(fileName);
+            var provider = new CombinedFileProvider(
+                new MockFileProvider(
+                    new MockFileInfo("FileA"),
+                    new MockFileInfo("FileB")
+                    ),
+                new MockFileProvider(
+                    expectedFileInfo,
+                    new MockFileInfo("File2")
+                    ),
+                new MockFileProvider(
+                    new MockFileInfo(fileName),
+                    new MockFileInfo("File3")
+                    )
+                );
+
+            // Act
+            var fileInfo = provider.GetFileInfo(fileName);
+
+            // Assert
+            Assert.NotNull(fileInfo);
```

Comment:
You could replace the three statements with `Assert.Same(expectedFileInfo, fileInfo);`